File: services/device/src/main/java/com/example/iter/device/dto/response/EquipmentDetailResponse.java
package com.example.iter.device.dto.response;

import com.example.iter.device.domain.entity.EquipmentCategory;
import com.example.iter.device.domain.entity.EquipmentStatus;
import com.example.iter.device.domain.entity.ProductConditionType;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

public record EquipmentDetailResponse(
        Long id,
        String name,
        EquipmentCategory category,
        String description,
        BigDecimal dailyPrice,
        LocalDate availableFrom,
        LocalDate availableTo,
        EquipmentStatus status,
        ProductConditionType productCondition,
        String conditionDetail,
        List<EquipmentImageResponse> images,
        EquipmentOwnerResponse owner,
        double averageRating,
        long reviewCount,
        LocalDateTime createdAt
) {
    public EquipmentDetailResponse {
        images = images == null ? List.of() : List.copyOf(images);
    }
}

File: services/device/src/main/kotlin/com/example/iter/device/dto/response/EquipmentDetailResponse.kt
package com.example.iter.device.dto.response

import com.example.iter.device.domain.entity.EquipmentCategory
import com.example.iter.device.domain.entity.EquipmentStatus
import com.example.iter.device.domain.entity.ProductConditionType

import java.math.BigDecimal
import java.time.LocalDate
import java.time.LocalDateTime

data class EquipmentDetailResponse(
    val id: Long,
    val name: String,
    val category: EquipmentCategory,
    val description: String?,
    val dailyPrice: BigDecimal,
    val availableFrom: LocalDate?,
    val availableTo: LocalDate?,
    val status: EquipmentStatus,
    val productCondition: ProductConditionType,
    val conditionDetail: String?,
    val images: List<EquipmentImageResponse>,
    val owner: EquipmentOwnerResponse,
    val averageRating: Double,
    val reviewCount: Long,
    val createdAt: LocalDateTime?,
)
